### DISCUSSION TITLE
Getting licence error when trying to create pivot table in FastAPI application using win32

### DISCUSSION THREAD

**Discussion by Divya-1477** [mhammond/pywin32#2525] [General]
Hi team,

I am facing issue while creating the pivot table in FastAPI application by using the win32. Getting the below error when     **pt_cache = wb.PivotCaches().Create(1, ws_data.Range("A1").CurrentRegion)** this line executing.

Error:  
pywintypes.com_error: (-2147352567, 'Exception occurred.', (0, 'Microsoft Excel', 'This command is unavailable because the license to use this application has expired.', 'xlmain11.chm', 0, -2146827284), None)

Please suggest me the way to resolve this.

Actual Requiremnt: Creating the Pivot table in excel through python FastAPI application with win32

**Comment by Avasam** [comment]
Doesn't this mean your Excel license is either expired or doesn't allow control through commands? Can you confirm and show that it is active?
It would be useful if you provided a complete reproduction too (all the code needed to trigger this issue, in a single script) as well as your Excel version.

A quick online search reveals that it could also be an issue on Excel's side, independent of license/product:
- https://forum.uipath.com/t/excel-application-scope-this-command-is-unavailable-because-the-license-to-use-this-application-has-expired/113261/2
- https://stackoverflow.com/questions/254207/license-expired-error-when-dynamically-generating-excel-docs-in-asp-net
- https://stackoverflow.com/questions/75435904/this-command-is-unavailable-because-the-license-to-use-this-application-has-expi
- https://www.mathworks.com/matlabcentral/answers/454552-how-do-i-stop-this-xlswrite-error-message
- https://forum.uipath.com/t/error-while-using-excel-application-scope/175870/7
